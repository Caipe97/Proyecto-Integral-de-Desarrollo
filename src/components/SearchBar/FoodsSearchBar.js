import React, { Component } from 'react';
import './SearchBar.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import DataTable from 'react-data-table-component';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSearch } from "@fortawesome/free-solid-svg-icons";
import CustomFoodModal from '../Modal/CustomFoodModal';
import Combobox from "react-widgets/Combobox";
import "react-widgets/styles.css";
import AddCategoryModal from '../Modal/AddCategoryModal';
import DeleteCategoryModal from '../Modal/DeleteCategoryModal';
import EditCategoryModal from '../Modal/EditCategoryModal';
import Add from '../../images/add.png';
import Delete from '../../images/delete.png';

const paginacionOpciones={
  rowsPerPageText: 'Filas por Página',
  rangeSeparatorText: 'de',
  selectAllRowsItem: true,
  selectAllRowsItemText: 'Todos',
  
}
const customStyles = {
	header: {
		style: {
			minHeight: '56px',
      background:'#CCDDE2',
      borderRadius:'17px',
      textShadow: '1px 1px black',
		},
	},
	headRow: {
		style: {
      backgroundColor: '#B6E052'
		},
	},
	headCells: {
		style: {
			'&:not(:last-of-type)': {},
		},
	},
  pagination: {
		style: {
			backgroundColor:'#B6E052'
		},
	},
  row:{
    pointer:{
      backgroundColor:'blue'
    }
  },
};
const conditionalRowStyles = [
    {
      when: row => row.caloriesPerServing > 0,
      style: {
        backgroundColor: '#B6E052',
        color: 'black',
        '&:hover': {
          cursor: 'pointer',
        },
      },
    },
  ];


class FoodsSearchBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      busquedaNombre: '',
      busquedaCategoria: '',
      alimentos: [],
      categorias: [],
      columnas:[],
      cantidadAlimentos: 0,
      cantidadCategorias: 0
    };
  }

  onChange = async event => {
    event.persist();
    await this.setState({busquedaNombre: event.target.value});
    this.filtrarElementos();
  }

  asignarColumnas=()=>{
    const columnas = [
      {
        cell:(row) => {
          if(row.userId){
            return(
              <div>
                <img src={Delete} alt='eliminar' onClick={() => {
                  const answer = window.confirm('Eliminar comida custom?');
                  if(answer){
                    this.props.onDeleteCustomFood(row.foodId)
                  }
                  }} id={row.foodId} style={{width: '20px', height: '20px', cursor: 'pointer'}}/>
                <CustomFoodModal edit={true} row={row} foodId={row.foodId} onEditCustomFood={this.props.onEditCustomFood} foodCategories={this.props.foodCategories}/>
                <img src={Add} alt='agregar' onClick={() => this.props.onAddFoodToCurrentMeal(row)} id={row.foodId} style={{width: '20px', height: '20px', cursor: 'pointer'}}/>
              </div>
            )
          } else {
            return(
            <img src={Add} alt='agregar' onClick={() => this.props.onAddFoodToCurrentMeal(row)} id={row.foodId} style={{width: '20px', height: '20px', cursor: 'pointer', marginLeft: 40}}/>
            )
          }
        }
        ,
        ignoreRowClick: true,
        allowOverflow: true,
        button: true
      },
      {
        name: 'Nombre',
        selector: row => row.name,
        sortable: true
      },
      {
        name: 'Categoría',
        selector: row => (
          <div>
            {
              this.props.foodCategories.map((foodCategory) => {
                if(row.foodCategoryId === foodCategory.foodCategoryId){
                  return(
                    <div key={foodCategory.foodCategoryId}>{foodCategory.name}</div>
                  )
                } else {
                  return(
                    <div key={foodCategory.foodCategoryId}></div>
                  )
                }
              })
            }
          </div>
        ),
        sortable: true
      },
      {
        name: 'Porción Recomendada',
        selector: row => row.recommendedServing,
        sortable: true
      },
      {
        name: 'Calorías Por Porción Recomendada',
        selector: row => row.caloriesPerServing,
        sortable: true
      }
    ];

    this.setState({columnas: columnas});
  }

  filtrarElementos=()=>{
    let search = this.props.foods.filter(food => {
      if(this.state.busquedaNombre === '' && this.state.busquedaCategoria === ''){
        return food;
      } else if(food.name.toLowerCase().includes(this.state.busquedaNombre)  && this.state.busquedaCategoria === ''){
        return food;
      } else if (this.state.busquedaNombre === '' && food.foodCategoryId.toString() === this.state.busquedaCategoria.toString()){
        return food;
      } else if(food.name.toLowerCase() === this.state.busquedaNombre && food.foodCategoryId.toString() === this.state.busquedaCategoria.toString()){
        return food;
      } else {
        return '';
      }
    });
    this.setState({alimentos: search});
  }

  async componentDidMount(){
    await this.props.onGetAllFoods(this.props.userId);
    await this.props.onGetFoodCategories(this.props.userId);
    let foodCategoriesCopy = [...this.props.foodCategories];
    foodCategoriesCopy.unshift({foodCategoryId: 0, name: 'Ver todos los alimentos', userId: null});
    this.setState({
      alimentos: this.props.foods,
      cantidadAlimentos: this.props.foods.length,
      categorias: foodCategoriesCopy,
      cantidadCategorias: this.props.foodCategories.length
    });
    this.asignarColumnas();
  }

  async componentDidUpdate(){
    if(this.props.foods.length !== this.state.cantidadAlimentos || this.props.foodCategories.length !== this.state.cantidadCategorias){
      await this.props.onGetAllFoods(this.props.userId);
      await this.props.onGetFoodCategories(this.props.userId);
      let foodCategoriesCopy = [...this.props.foodCategories];
      foodCategoriesCopy.unshift({foodCategoryId: 0, name: 'Ver todos los alimentos', userId: null});
      this.setState({
        alimentos: this.props.foods,
        cantidadAlimentos: this.props.foods.length,
        categorias: foodCategoriesCopy,
        cantidadCategorias: this.props.foodCategories.length
      });
      this.asignarColumnas();
    }
  }
  
  onChangeComboBox = async event => {
    if(event.foodCategoryId === 0){ //Ver todos los alimentos
      await this.setState({...this.state, busquedaCategoria: ''});
    } else if(event.foodCategoryId === -1){ //Crear un nuevo alimento
      await this.setState({...this.state, busquedaCategoria: ''});
      console.log('ABRIR MODAL');
    } else {
      await this.setState({...this.state, busquedaCategoria: event.foodCategoryId});
    }
    this.filtrarElementos();
  }

  render(){
    return (
      <div className="table-responsive" style={{backgroundColor:'#B6E052'}}>
        <div className="barraBusquedaNombre" style={{backgroundColor:'#B6E052'}}>
          <div style={{backgroundColor: 'lightblue', borderRadius: 10, display: 'flex',justifyContent: 'space-between', alignContent: 'center', alignItems: 'center', marginBottom: 20}}>
            <h1 style={{color: 'rgb(0, 38, 38)', fontFamily: 'Arial', fontSize: 30, alignSelf: 'center', textAlign: 'center', marginTop: 8, marginLeft: 5}}>Alimentos</h1>
            <div>
              <input
              type="text"
              placeholder="Buscar por nombre"
              className="textField"
              name="busquedaNombre"
              value={this.state.busquedaNombre}
              onChange={this.onChange}
              style={{borderRadius:'13px'}}
              />
              <button type="button" className="btnBuscar" style={{borderRadius:'19px'}}>
              {" "}
              <FontAwesomeIcon icon={faSearch} />
              </button>
            </div>
          </div>
          <div style={{display: 'flex', justifyContent: 'space-between', alignItems: 'center'}}>
            <Combobox
              data={this.state.categorias}
              defaultValue={this.state.categorias[0]}
              textField='name'
              onSelect={this.onChangeComboBox}
              groupBy={category => category.userId}
              placeholder= "Buscar por categoría"
              renderListGroup={ ({group}) => ( //group es el userId
                <span>
                  {group === null ? 'Default' : 'Custom'}
                </span>
              )}
              style={{width: "80%"}}
              />
            <div style={{paddingLeft: 10, width: "20%", justifyContent: 'space-between'}}>
              <AddCategoryModal userId={this.props.userId} onCreateCategory={this.props.onCreateCategory} isPending={this.props.foodCategoryIsPending}/>
              <DeleteCategoryModal userId={this.props.userId} onDeleteCategory={this.props.onDeleteCategory} foodCategories={this.props.foodCategories} isPending={this.props.foodCategoryIsPending}/>
              <EditCategoryModal userId={this.props.userId} onEditCategory={this.props.onEditCategory} foodCategories={this.props.foodCategories} isPending={this.props.foodCategoryIsPending}/>
            </div>
          </div>
          
        </div>
        <DataTable 
          columns={this.state.columnas}
          data={this.state.alimentos}
          pagination
          paginationComponentOptions={paginacionOpciones}
          fixedHeader
          fixedHeaderScrollHeight="600px"
          noDataComponent={
            <span style={{marginTop: 150, marginBottom: 150}}>No se encontró ningún alimento</span>
          }
          conditionalRowStyles={conditionalRowStyles}
          customStyles={customStyles}
          pointer
        />
      </div>
    );
  }
}
export default FoodsSearchBar;