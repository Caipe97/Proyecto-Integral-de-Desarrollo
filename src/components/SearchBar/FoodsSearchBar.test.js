import React from 'react';
import { shallow } from 'enzyme';
import FoodsSearchBar from './FoodsSearchBar';

describe('FoodsSearchBar tests', () => {
    const historyMock = { push: jest.fn() };
    const mockProps1 =  {
        userId: 0, 
        onLogout: jest.fn(),
        userId: 1,state : {
            busquedaNombre: '',
            busquedaCategoria: 1,
            alimentos: [],
            categorias: [],
            columnas:[],
            cantidadAlimentos: 3,
            cantidadCategorias: 0
          },
            meals:[
                {mealId: 21, name: 'queso', foodsAndQuantity: {quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
                {mealId: 22, name: 'carne', foodsAndQuantity: {quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
                {mealId: 23, name: 'jamon', foodsAndQuantity: {quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1}
            ],
            meal: {
                FoodList:[
                    {quantity: 1, food: {foodId: 1, name: 'Milanesa', 
                    recommendedServing: 85, caloriesPerServing: 198, 
                    createdAt: '2021-09-15T19:58:04.486Z'}}, 
                ]
            },
            foods: [
                {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'},
                {foodId: 2, name: 'Papa', recommendedServing: 200, caloriesPerServing: 100, createdAt: '2021-09-15T20:02:16.490Z'}
            ],
                foodCategories:[
                        {
                          foodCategoryId: 1,
                          name: "Fruta",
                          createdAt: "2021-09-30T14:09:34.756Z",
                          updatedAt: "2021-09-30T14:09:34.756Z",
                          userId: null
                        },
                        {
                          foodCategoryId: 7,
                          name: "Fideos",
                          createdAt: "2021-09-30T18:58:58.542Z",
                          updatedAt: "2021-09-30T18:58:58.542Z",
                          userId: null
                        }
                      ],
                      foods: [
                        {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'},
                        {foodId: 2, name: 'Papa', recommendedServing: 200, caloriesPerServing: 100, createdAt: '2021-09-15T20:02:16.490Z'}
                    ],
        onAddMeal: jest.fn(),
        onGetMealsFromUser:jest.fn(),
        onDeleteMeal: jest.fn(),
        onGetAllFoods:jest.fn(),
        onAddFoodToCurrentMeal: jest.fn(),
        onRemoveFoodFromCurrentMeal:jest.fn(),
        onLogout: jest.fn(),
        onResetCurrentMeal: jest.fn(),
        onUpdateCurrentMeal:jest.fn(),
        onChangeCurrentMealNameAndDateEaten:jest.fn(),    
        onAddCustomFood:jest.fn(), 
        onEditCustomFood:jest.fn(),
        onDeleteCustomFood:jest.fn(),
        onGetFoodCategories: jest.fn(),
        onCreateCategory: jest.fn(),
        onDeleteCategory: jest.fn(),
        onEditCategory: jest.fn(),
        preventDefault:  jest.fn()
    };
    let wrapper,wrapper2;
    const mockProps2 = {
        userId: 0, 
        onLogout: jest.fn(),
        userId: 1,state : {
            busquedaNombre: '',
            busquedaCategoria: '',
            alimentos: [],
            categorias: [],
            columnas:[],
            cantidadAlimentos: 0,
            cantidadCategorias: 0
          },
            meals:[
                {mealId: 21, name: 'queso', foodsAndQuantity: {quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
                {mealId: 22, name: 'carne', foodsAndQuantity: {quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1},
                {mealId: 23, name: 'jamon', foodsAndQuantity: {quantity: 1, food: {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'}}, dateEaten: '2021-09-09T00:00:00.000Z', userId: 1}
            ],
            meal: {
                FoodList:[
                    {quantity: 1, food: {foodId: 1, name: 'Milanesa', 
                    recommendedServing: 85, caloriesPerServing: 198, 
                    createdAt: '2021-09-15T19:58:04.486Z'}}, 
                ]
            },
            foods: [
                {foodId: 1, name: 'Milanesa', recommendedServing: 85, caloriesPerServing: 198, createdAt: '2021-09-15T19:58:04.486Z'},
                {foodId: 2, name: 'Papa', recommendedServing: 200, caloriesPerServing: 100, createdAt: '2021-09-15T20:02:16.490Z'}
            ],
                foodCategories:[
                        {
                          foodCategoryId: 1,
                          name: "Fruta",
                          createdAt: "2021-09-30T14:09:34.756Z",
                          updatedAt: "2021-09-30T14:09:34.756Z",
                          userId: null
                        },
                        {
                          foodCategoryId: 7,
                          name: "Fideos",
                          createdAt: "2021-09-30T18:58:58.542Z",
                          updatedAt: "2021-09-30T18:58:58.542Z",
                          userId: null
                        }
                      ],
        
        onAddMeal: jest.fn(),
        onGetMealsFromUser:jest.fn(),
        onDeleteMeal: jest.fn(),
        onGetAllFoods:jest.fn(),
        onAddFoodToCurrentMeal: jest.fn(),
        onRemoveFoodFromCurrentMeal:jest.fn(),
        onLogout: jest.fn(),
        onResetCurrentMeal: jest.fn(),
        onUpdateCurrentMeal:jest.fn(),
        onChangeCurrentMealNameAndDateEaten:jest.fn(),    
        onAddCustomFood:jest.fn(), 
        onEditCustomFood:jest.fn(),
        onDeleteCustomFood:jest.fn(),
        onGetFoodCategories: jest.fn(),
        onCreateCategory: jest.fn(),
        onDeleteCategory: jest.fn(),
        onEditCategory: jest.fn(),
        preventDefault:  jest.fn()
    };

    beforeEach(() => {
      wrapper = shallow(<FoodsSearchBar {...mockProps2} />);
      wrapper2 = shallow(<FoodsSearchBar {...mockProps1} />);
    })

    it('expect to render FoodsSearchBar component to location state ', async() => {
      expect(wrapper).toMatchSnapshot();//if
      await wrapper.instance().componentDidMount();
    })
    it('expect to render FoodsSearchBar component to location state2 ', async() => {
      expect(wrapper2).toMatchSnapshot();//if
      await wrapper2.instance().componentDidMount();
      await wrapper2.instance().componentDidUpdate();
    })
    it('OnChange expect t11', async () => {
      await wrapper.find('[type="button"]').at(0).simulate('click',jest.fn);
      wrapper.find('[type="button"]').at(0).simulate('click',{ persist: jest.fn()});
    })
    it('OnChange should because name contains search value not return food and return all foods because busquedaNombre is null',async() => {
      await wrapper.instance().componentDidMount();
      await wrapper.find('input').at(0).simulate('change', { persist: jest.fn(),target: { name: 'busquedaNombre', value: '' }});
       
    })
    it('OnChange should because name contains search value return null because dont find value ',async() => {
      await wrapper.instance().componentDidMount();
      await wrapper.find('input').at(0).simulate('change', { persist: jest.fn(),target: { name: 'busquedaNombre', value: 'Plan febrero' }});
       
    })
    it('OnChange should because name contains search value return food when foodname includes in busquedaNombre',async() => {
      await wrapper.instance().componentDidMount();
      await wrapper.find('input').at(0).simulate('change', { persist: jest.fn(),target: { name: 'busquedaNombre', value: 'milanesa' }});
     
    })
    it('OnChange should because name contains search value not return food and return all foods with categoryID define because busquedaNombre is null ',async() => {
      await wrapper2.instance().componentDidMount();
      await wrapper2.find('input').at(0).simulate('change', { persist: jest.fn(),target: { name: 'busquedaNombre', value: 'milanesa' }});
       
    })
    it('OnChange should because name contains search value not return food and return all foods with categoryID define because busquedaNombre is null ',async() => {
      await wrapper2.instance().componentDidMount();
      await wrapper2.find('input').at(0).simulate('change', { persist: jest.fn(),target: { name: 'busquedaNombre', value: '' }});
       
    })
})
