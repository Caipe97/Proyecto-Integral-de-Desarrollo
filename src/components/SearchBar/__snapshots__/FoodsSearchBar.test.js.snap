// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodsSearchBar tests expect to render FoodsSearchBar component to location state  1`] = `
<div
  className="table-responsive"
  style={
    Object {
      "backgroundColor": "#B6E052",
    }
  }
>
  <div
    className="barraBusquedaNombre"
    style={
      Object {
        "backgroundColor": "#B6E052",
      }
    }
  >
    <div
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "lightblue",
          "borderRadius": 10,
          "display": "flex",
          "justifyContent": "space-between",
          "marginBottom": 20,
        }
      }
    >
      <h1
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(0, 38, 38)",
            "fontFamily": "Arial",
            "fontSize": 30,
            "marginLeft": 5,
            "marginTop": 8,
            "textAlign": "center",
          }
        }
      >
        Alimentos
      </h1>
      <div>
        <input
          className="textField"
          name="busquedaNombre"
          onChange={[Function]}
          placeholder="Buscar por nombre"
          style={
            Object {
              "borderRadius": "13px",
            }
          }
          type="text"
          value=""
        />
        <button
          className="btnBuscar"
          style={
            Object {
              "borderRadius": "19px",
            }
          }
          type="button"
        >
           
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f002",
                  "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                ],
                "iconName": "search",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <Combobox
        data={
          Array [
            Object {
              "foodCategoryId": 0,
              "name": "Ver todos los alimentos",
              "userId": null,
            },
            Object {
              "createdAt": "2021-09-30T14:09:34.756Z",
              "foodCategoryId": 1,
              "name": "Fruta",
              "updatedAt": "2021-09-30T14:09:34.756Z",
              "userId": null,
            },
            Object {
              "createdAt": "2021-09-30T18:58:58.542Z",
              "foodCategoryId": 7,
              "name": "Fideos",
              "updatedAt": "2021-09-30T18:58:58.542Z",
              "userId": null,
            },
          ]
        }
        defaultValue={
          Object {
            "foodCategoryId": 0,
            "name": "Ver todos los alimentos",
            "userId": null,
          }
        }
        groupBy={[Function]}
        onSelect={[Function]}
        placeholder="Buscar por categoría"
        renderListGroup={[Function]}
        style={
          Object {
            "width": "80%",
          }
        }
        textField="name"
      />
      <div
        style={
          Object {
            "justifyContent": "space-between",
            "paddingLeft": 10,
            "width": "20%",
          }
        }
      >
        <AddCategoryModal
          onCreateCategory={[MockFunction]}
          userId={1}
        />
        <DeleteCategoryModal
          categories={
            Array [
              Object {
                "createdAt": "2021-09-30T14:09:34.756Z",
                "foodCategoryId": 1,
                "name": "Fruta",
                "updatedAt": "2021-09-30T14:09:34.756Z",
                "userId": null,
              },
              Object {
                "createdAt": "2021-09-30T18:58:58.542Z",
                "foodCategoryId": 7,
                "name": "Fideos",
                "updatedAt": "2021-09-30T18:58:58.542Z",
                "userId": null,
              },
            ]
          }
          onDeleteCategory={[MockFunction]}
          userId={1}
        />
        <EditCategoryModal
          categories={
            Array [
              Object {
                "createdAt": "2021-09-30T14:09:34.756Z",
                "foodCategoryId": 1,
                "name": "Fruta",
                "updatedAt": "2021-09-30T14:09:34.756Z",
                "userId": null,
              },
              Object {
                "createdAt": "2021-09-30T18:58:58.542Z",
                "foodCategoryId": 7,
                "name": "Fideos",
                "updatedAt": "2021-09-30T18:58:58.542Z",
                "userId": null,
              },
            ]
          }
          onEditCategory={[MockFunction]}
          userId={1}
        />
      </div>
    </div>
  </div>
  <Memo()
    columns={
      Array [
        Object {
          "allowOverflow": true,
          "button": true,
          "cell": [Function],
          "ignoreRowClick": true,
        },
        Object {
          "name": "Nombre",
          "selector": [Function],
          "sortable": true,
        },
        Object {
          "name": "Categoría",
          "selector": [Function],
          "sortable": true,
        },
        Object {
          "name": "Porción Recomendada",
          "selector": [Function],
          "sortable": true,
        },
        Object {
          "name": "Calorías Por Porción Recomendada",
          "selector": [Function],
          "sortable": true,
        },
      ]
    }
    conditionalRowStyles={
      Array [
        Object {
          "style": Object {
            "&:hover": Object {
              "cursor": "pointer",
            },
            "backgroundColor": "#B6E052",
            "color": "black",
          },
          "when": [Function],
        },
      ]
    }
    customStyles={
      Object {
        "headCells": Object {
          "style": Object {
            "&:not(:last-of-type)": Object {},
          },
        },
        "headRow": Object {
          "style": Object {
            "backgroundColor": "#B6E052",
          },
        },
        "header": Object {
          "style": Object {
            "background": "#CCDDE2",
            "borderRadius": "17px",
            "minHeight": "56px",
            "textShadow": "1px 1px black",
          },
        },
        "pagination": Object {
          "style": Object {
            "backgroundColor": "#B6E052",
          },
        },
        "row": Object {
          "pointer": Object {
            "backgroundColor": "blue",
          },
        },
      }
    }
    data={
      Array [
        Object {
          "caloriesPerServing": 198,
          "createdAt": "2021-09-15T19:58:04.486Z",
          "foodId": 1,
          "name": "Milanesa",
          "recommendedServing": 85,
        },
        Object {
          "caloriesPerServing": 100,
          "createdAt": "2021-09-15T20:02:16.490Z",
          "foodId": 2,
          "name": "Papa",
          "recommendedServing": 200,
        },
      ]
    }
    fixedHeader={true}
    fixedHeaderScrollHeight="600px"
    noDataComponent={
      <span
        style={
          Object {
            "marginBottom": 150,
            "marginTop": 150,
          }
        }
      >
        No se encontró ningún alimento
      </span>
    }
    pagination={true}
    paginationComponentOptions={
      Object {
        "rangeSeparatorText": "de",
        "rowsPerPageText": "Filas por Página",
        "selectAllRowsItem": true,
        "selectAllRowsItemText": "Todos",
      }
    }
    pointer={true}
  />
</div>
`;

exports[`FoodsSearchBar tests expect to render FoodsSearchBar component to location state2  1`] = `
<div
  className="table-responsive"
  style={
    Object {
      "backgroundColor": "#B6E052",
    }
  }
>
  <div
    className="barraBusquedaNombre"
    style={
      Object {
        "backgroundColor": "#B6E052",
      }
    }
  >
    <div
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "backgroundColor": "lightblue",
          "borderRadius": 10,
          "display": "flex",
          "justifyContent": "space-between",
          "marginBottom": 20,
        }
      }
    >
      <h1
        style={
          Object {
            "alignSelf": "center",
            "color": "rgb(0, 38, 38)",
            "fontFamily": "Arial",
            "fontSize": 30,
            "marginLeft": 5,
            "marginTop": 8,
            "textAlign": "center",
          }
        }
      >
        Alimentos
      </h1>
      <div>
        <input
          className="textField"
          name="busquedaNombre"
          onChange={[Function]}
          placeholder="Buscar por nombre"
          style={
            Object {
              "borderRadius": "13px",
            }
          }
          type="text"
          value=""
        />
        <button
          className="btnBuscar"
          style={
            Object {
              "borderRadius": "19px",
            }
          }
          type="button"
        >
           
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  512,
                  512,
                  Array [],
                  "f002",
                  "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
                ],
                "iconName": "search",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </button>
      </div>
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <Combobox
        data={
          Array [
            Object {
              "foodCategoryId": 0,
              "name": "Ver todos los alimentos",
              "userId": null,
            },
            Object {
              "createdAt": "2021-09-30T14:09:34.756Z",
              "foodCategoryId": 1,
              "name": "Fruta",
              "updatedAt": "2021-09-30T14:09:34.756Z",
              "userId": null,
            },
            Object {
              "createdAt": "2021-09-30T18:58:58.542Z",
              "foodCategoryId": 7,
              "name": "Fideos",
              "updatedAt": "2021-09-30T18:58:58.542Z",
              "userId": null,
            },
          ]
        }
        defaultValue={
          Object {
            "foodCategoryId": 0,
            "name": "Ver todos los alimentos",
            "userId": null,
          }
        }
        groupBy={[Function]}
        onSelect={[Function]}
        placeholder="Buscar por categoría"
        renderListGroup={[Function]}
        style={
          Object {
            "width": "80%",
          }
        }
        textField="name"
      />
      <div
        style={
          Object {
            "justifyContent": "space-between",
            "paddingLeft": 10,
            "width": "20%",
          }
        }
      >
        <AddCategoryModal
          onCreateCategory={[MockFunction]}
          userId={1}
        />
        <DeleteCategoryModal
          categories={
            Array [
              Object {
                "createdAt": "2021-09-30T14:09:34.756Z",
                "foodCategoryId": 1,
                "name": "Fruta",
                "updatedAt": "2021-09-30T14:09:34.756Z",
                "userId": null,
              },
              Object {
                "createdAt": "2021-09-30T18:58:58.542Z",
                "foodCategoryId": 7,
                "name": "Fideos",
                "updatedAt": "2021-09-30T18:58:58.542Z",
                "userId": null,
              },
            ]
          }
          onDeleteCategory={[MockFunction]}
          userId={1}
        />
        <EditCategoryModal
          categories={
            Array [
              Object {
                "createdAt": "2021-09-30T14:09:34.756Z",
                "foodCategoryId": 1,
                "name": "Fruta",
                "updatedAt": "2021-09-30T14:09:34.756Z",
                "userId": null,
              },
              Object {
                "createdAt": "2021-09-30T18:58:58.542Z",
                "foodCategoryId": 7,
                "name": "Fideos",
                "updatedAt": "2021-09-30T18:58:58.542Z",
                "userId": null,
              },
            ]
          }
          onEditCategory={[MockFunction]}
          userId={1}
        />
      </div>
    </div>
  </div>
  <Memo()
    columns={
      Array [
        Object {
          "allowOverflow": true,
          "button": true,
          "cell": [Function],
          "ignoreRowClick": true,
        },
        Object {
          "name": "Nombre",
          "selector": [Function],
          "sortable": true,
        },
        Object {
          "name": "Categoría",
          "selector": [Function],
          "sortable": true,
        },
        Object {
          "name": "Porción Recomendada",
          "selector": [Function],
          "sortable": true,
        },
        Object {
          "name": "Calorías Por Porción Recomendada",
          "selector": [Function],
          "sortable": true,
        },
      ]
    }
    conditionalRowStyles={
      Array [
        Object {
          "style": Object {
            "&:hover": Object {
              "cursor": "pointer",
            },
            "backgroundColor": "#B6E052",
            "color": "black",
          },
          "when": [Function],
        },
      ]
    }
    customStyles={
      Object {
        "headCells": Object {
          "style": Object {
            "&:not(:last-of-type)": Object {},
          },
        },
        "headRow": Object {
          "style": Object {
            "backgroundColor": "#B6E052",
          },
        },
        "header": Object {
          "style": Object {
            "background": "#CCDDE2",
            "borderRadius": "17px",
            "minHeight": "56px",
            "textShadow": "1px 1px black",
          },
        },
        "pagination": Object {
          "style": Object {
            "backgroundColor": "#B6E052",
          },
        },
        "row": Object {
          "pointer": Object {
            "backgroundColor": "blue",
          },
        },
      }
    }
    data={
      Array [
        Object {
          "caloriesPerServing": 198,
          "createdAt": "2021-09-15T19:58:04.486Z",
          "foodId": 1,
          "name": "Milanesa",
          "recommendedServing": 85,
        },
        Object {
          "caloriesPerServing": 100,
          "createdAt": "2021-09-15T20:02:16.490Z",
          "foodId": 2,
          "name": "Papa",
          "recommendedServing": 200,
        },
      ]
    }
    fixedHeader={true}
    fixedHeaderScrollHeight="600px"
    noDataComponent={
      <span
        style={
          Object {
            "marginBottom": 150,
            "marginTop": 150,
          }
        }
      >
        No se encontró ningún alimento
      </span>
    }
    pagination={true}
    paginationComponentOptions={
      Object {
        "rangeSeparatorText": "de",
        "rowsPerPageText": "Filas por Página",
        "selectAllRowsItem": true,
        "selectAllRowsItemText": "Todos",
      }
    }
    pointer={true}
  />
</div>
`;
